### DISCUSSION TITLE
No SmallRye Stork ConfigProvider found

### DISCUSSION THREAD

**Discussion by eichstaedtk** [smallrye/smallrye-stork#191] [Q&A]
Hi There, 

Iam using Quarkus 2.6.1 and trying to integrate Stork and failed. The first thing iam not sure ist that I using YAML config files instead of properties. 

Is this an problem using Stork with Quarkus? 

The second Thing is that I getting the following error during Stork initialization: 

No SmallRye Stork ConfigProvider found

Can maybe someone help me? 

Best regards 

Konrad

**Comment by aureamunoz** [comment]
Hi @eichstaedtk! 

Thanks for creating this, I've just moved the issue to a `Q&A` discussion because it doesn't seem to be a bug, at least not yet ;-).

Could you please share the yaml file that you are using? YAML files are supported so this shouldn't be a problem. 
The `No SmallRye Stork ConfigProvider found` is telling that it doesn't find a config provider. Check that you have the following Quarkus extensions in your project: 
```
quarkus-rest-client-reactive
quarkus-resteasy-reactive
```
I let you this article that could help you: https://quarkus.io/blog/smallrye-stork-intro/#stork-and-quarkus

**Reply by eichstaedtk** [reply]
Many Thanks for your support. After I added the following depedency to my project I got one step further


```
 <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-smallrye-stork</artifactId>
    </dependency>
```

Now the No SmallRye Stork ConfigProvider found disappears but I got two problems: 

1.) First my serviceregistry is eureka and not localted under the root path / but under /serviceregistry. The problem ist that your eureka config has no possibility to configure this because only host and port are avaliable. What can I do? 

2.) The second thing ist that Iam using quarkus-rest.client instead of quarkus-rest-client-reactive. In this case I got the following error: 
java.lang.IllegalArgumentException: The value of URL was invalid stork://IMPORTSERVICE/rest
	at io.quarkus.restclient.runtime.RestClientBase.configureBaseUrl(RestClientBase.java:307)
	at io.quarkus.restclient.runtime.RestClientBase.create(RestClientBase.java:58)
....
Caused by: java.net.MalformedURLException: unknown protocol: stork
	at java.base/java.net.URL.<init>(URL.java:634)

I there any possibility to get this work using quarkus-rest-client? 

Many Thanks , Konrad

**Reply by aureamunoz** [reply]
Hi again Konrad,

1) I see, you have a custom context path in the Eureka server config. We would need to add this also as a property config in Stork Eureka service discovery, could you please create an issue for this?

2)You should use the Quarkus REST Client Reactive if you want built-in integration with Stork.

I hope this helps

**Reply by eichstaedtk** [reply]
Hi Aurea, 

Many Thanks for your support, I saw you already open an issue for the Eureka Context Path. I will try using Rest Client Reactive. 

Best regards 

Konrad
